starter-code/scripts/models/article.js

```diff
@@ -56,7 +60,16 @@ Article.fetchAll = function() {
      1. Load our json data,
      2. Store that data in localStorage so we can skip the server call next time.
      3. And then render the index page. */
+    $.getJSON('../data/hackerIpsum.json',function(data){
+      localStorage.hackerIpsum = JSON.stringify(data);
+      Article.loadAll(data);
+      articleView.renderIndexPage();
+    });
+
+    // localStorage.hackerIpsum = JSON.stringify(newData);
+
   }
```

Comment:
We tried that, and the problem was that when we called `Article.loadAll()` and `articleView.renderIndexPage()'
outside of the `if/else` block, when we had to load the data from the file (the `else` condition), the data wouldn't finish loading before the two functions were called, so we'd get no articles. So we had to put the functions into the callback for the `getJSON` so that the data would load completely before we tried to call those functions.